src/drivers/mongodb.ts

```diff
@@ -0,0 +1,80 @@
+import { defineDriver } from "./utils";
+import { Collection, MongoClient } from "mongodb";
+
+export interface MongoDbOptions {
+  /**
+   * The MongoDB connection string.
+   */
+  connectionString?: string;
+  /**
+   * The name of the database to use.
+   * @default "unstorage"
+   */
+  databaseName?: string;
+  /**
+   * The name of the collection to use.
+   * @default "unstorage"
+   */
+  collectionName?: string;
+}
+
+export default defineDriver((opts: MongoDbOptions = {}) => {
+  const {
+    connectionString,
+    databaseName = "unstorage",
+    collectionName = "unstorage",
+  } = opts;
+  if (!connectionString)
+    throw new Error(
+      "MongoDb driver requires a connection string to be provided."
+    );
+  let client: Collection;
+  const getMongoDbClient = () => {
+    if (!client) {
+      const mongoClient = new MongoClient(connectionString);
+      const db = mongoClient.db(databaseName);
+      client = db.collection(collectionName);
+    }
+    return client;
+  };
+
+  return {
+    async hasItem(key) {
+      const result = await getMongoDbClient().findOne({ key });
+      return !!result;
+    },
+    async getItem(key) {
+      const document = await getMongoDbClient().findOne({ key });
+      return document?.value ? document.value : null;
+    },
+    async setItem(key, value) {
+      const currentDateTime = new Date();
+      const [modified, created] = [currentDateTime, currentDateTime];
+      await getMongoDbClient().findOneAndUpdate(
+        { key },
+        { $set: { key, value, modified }, $setOnInsert: { created } },
+        { upsert: true, returnDocument: "after" }
+      );
+      return;
+    },
+    async removeItem(key) {
+      await getMongoDbClient().deleteOne({ key });
+    },
+    async getKeys() {
+      const documents = await getMongoDbClient().find().toArray();
+      const keys = [];
+      documents.map((document) => keys.push(document.key));
+      return keys;
```

Comment:
What about:

```suggestion
      return await getMongoDbClient()
        .find()
        .map((document) => document.key)
        .toArray();
```

Reply:
Passes all tests, I think I thought it would return an iterator, but it returns a cursor so this is a much better way 👍